lib/prelude.dx

```diff
@@ -244,10 +243,10 @@ def clamp_nonneg_prim (a: Int) : Int =
   %select isNeg z a
 
 -- TODO: Define Range in surface syntax
-instance {l h} Ix (Range l h)
-  size = clamp_nonneg_prim $ h - l
+instance {n} Ix (Fin n)
+  size = n
   ordinal = \i. %cast Int i
-  unsafe_from_ordinal = \i. %cast (Range l h) i
+  unsafe_from_ordinal = \i. %cast (Fin n) i
```

Comment:
Do we still need this instance? I think that dict synthesis has a special path for `Ix (Fin n)` now, doesn't it?

Reply:
And if we don't need it, then we should also make sure that we report the ambiguity instead of silently ignoring any user-defined instances

Reply:
Good point. Let's remove it. I just tried that and exposed a bug in my special path for `Ix (Fin n)`. It does the check at the entry point to synthesis rather than internally (i.e. for dictionaries required by other dictionaries). Just pushed the fix.